unit-tests/internal/internal-tests-extrinsic.cpp

```diff
@@ -12,22 +12,87 @@
-inline void require_zero_vector(const float(&vector)[3])
-{
-    for (int i = 1; i < 3; ++i) REQUIRE(vector[i] == 0.0f);
-}
+constexpr int ITERATIONS_PER_CONFIG = 100;
+constexpr int INNER_ITERATIONS_PER_CONFIG = 10;
+constexpr int DELAY_INCREMENT_THRESHOLD = 3; //[%]
+constexpr int DELAY_INCREMENT_THRESHOLD_IMU = 8; //[%]
+constexpr int SPIKE_THRESHOLD = 2; //[stdev]
 
-// Require that matrix is exactly the identity matrix
-inline void require_identity_matrix(const float(&matrix)[9])
+// Input:     vector that represent samples of delay to first frame of one stream
+// Output:  - vector of line fitting data according to least squares algorithm 
+//          - slope of the fitted line
+// reference: https://en.wikipedia.org/wiki/Least_squares
+double line_fitting(const std::vector<double>& y_vec, std::vector<double>& y_fit = std::vector<double>())
 {
-    static const float identity_matrix_3x3[] = { 1,0,0, 0,1,0, 0,0,1 };
-    for (int i = 0; i < 9; ++i) REQUIRE(matrix[i] == approx(identity_matrix_3x3[i]));
+    double ysum = std::accumulate(y_vec.begin(), y_vec.end(), 0.0);  //calculate sigma(yi)
+    double xsum = 0;
+    double x2sum = 0;
+    double xysum = 0;
+    auto y_vec_it = y_vec.begin();
+    size_t n = y_vec.size();
+    for (auto i = 0; i < n; i++)
+    {
+        xsum = xsum + i;  //sigma(xi)                                   
+        x2sum = x2sum + pow(i, 2); //sigma(x^2i)
+        xysum = xysum + i * *(y_vec_it + i);
+    }
+    double a = (n * xysum - xsum * ysum) / (n * x2sum - xsum * xsum);      //slope
+    double b = (x2sum * ysum - xsum * xysum) / (x2sum * n - xsum * xsum);  //intercept   
+    if (y_fit.size() > 0)
+    {
+        auto it = y_fit.begin();
+        for (auto i = 0; i < n; i++)
+        {
+            *(it + i) = a * i + b;
+        }
+    }
+    return a; // slope is used when checking delay increment 
 }
+// Input:  vector that represent samples of time delay to first frame of one stream
+// Output: vector of filtered-out spikes 
+void data_filter(const std::vector<double>& stream_vec, std::vector<double>& filtered_stream_vec)
+{
+    std::vector<double> y_fit(stream_vec.size());
+    double slope = line_fitting(stream_vec, y_fit);
+
+    auto y_fit_it = y_fit.begin();
+    auto stream_vec_it = stream_vec.begin();
+    std::vector<double> diff_y_fit;
+    for (auto i = 0; i < stream_vec.size(); i++)
+    {
+        double diff = abs(*(y_fit_it + i) - *(stream_vec_it + i));
+        diff_y_fit.push_back(diff);
+    }
+    // calc stdev from fitted linear line
+    double sq_sum = std::inner_product(diff_y_fit.begin(), diff_y_fit.end(), diff_y_fit.begin(), 0.0);
+    double stdev = std::sqrt(sq_sum / diff_y_fit.size());
 
+    // calc % of each sample from stdev
+    std::vector<double> samples_stdev(diff_y_fit.size());
+    auto v_size = diff_y_fit.size();
+    std::transform(diff_y_fit.begin(), diff_y_fit.end(), samples_stdev.begin(), [stdev](double d) {
+        d = d < 0 ? -d : d;
```

Comment:
Pls check indentation
`d` is positive by design, the extra tests here in L83 can be skipped

Reply:
indeed it is positive because I took abs when calculatin samples_stdev vector